external/nyc311/models.py

```diff
@@ -0,0 +1,12 @@
+import attr
+
+
+@attr.s
+class NYC311Complaint(object):
```

Comment:
I've read some articles about attr, it looks like a package can make your code compact!
Could you provide some reason/practice about using it on the model?

Reply:
Basically I wanted to use it as a replacement of dictionary. I think sometimes this kind of data class can work as a documented version of a dictionary.

And yes it's kind of verbose that you write this kind of class in python. We need to create `__init__ `with parameters and a bunch of `self.XXX = XXX`. That's why I used attrs for the model.

In case we need to store that model, we'll use django model which is backed by ORM.

Reply:
I'm considering specifying the type of the data!

Reply:
Do you mean we are going to create a db schema without using theirs?

Reply:
No I meant `attrs` supports type annotation and also I found I was wrong. attrs seems to not support type annotation but it supports converter. So I was considering specifying converter but not sure it might worth.
```
>>> @attr.s
... class C(object):
...     x = attr.ib(converter=int)
>>> o = C("1")
>>> o.x
1
```

Reply:
@alldne I'm also not sure about this too. But it seems dangerous to me.
So, in the example, "1" is supported to convert.
however, if the input becomes "abc", will throw ValueError.
Just keep in mind about this potential error, I don't want to break the current prepared for demo.

Reply:
Thanks for pointing out! Honestly, we cannot make sure this kind of integrity perfectly in python because it's dynamically typed language. I agree it's still dangerous when we specify converter.

By the way I'd like to point out that the problem also exists when we use django model or just raw dictionary. By the problem I meant a risk of runtime crash. In other words, the purpose of these changes are not to prevent those kind of type mismatch, but to let us know what a data looks like and possibly pycharm or vscode can provide us some neat autocompletion!

For the runtime crash, I think it's matter of when this erroneous cases pop out. We cannot find this during complie time because it's python. Instead we write tests. If we have good test coverage, we can find out early.

Reply:
@alldne So for using django model, when I try to enter a string that contains not only numbers in the database, it will prevent me to do so. Therefore, for me, it's a kind of prevention for unexpected data.